src/main/java/net/snowflake/client/core/SFFixedViewResultSet.java

```diff
@@ -129,6 +129,7 @@ public SFStatementType getStatementType() {
 
   @Override
   public void setStatementType(SFStatementType statementType) throws SQLException {
+    SnowflakeSQLLoggedException.logSqlFeatureNotSupportedException(session);
     throw new SQLFeatureNotSupportedException();
```

Comment:
can `SnowfalkeFeatureNotSupportedException` wrap both `SnowflakeSQLLoggedException.logSqlFeatureNotSupportedException(session)` and `SQLFeatureNotSupportedException`?

Reply:
I could make a separate class **SnowflakeSqlFeatureNotSupportedException** as an extension of SQLFeatureNotSupportedException and put this function in there, and then put the telemetry logging functions in a class that can be accessed by both SnowflakeSqlFeatureNotSupportedException and SnowflakeSQLLoggedException. They can't be in the same class because they extend different types of exceptions. Do you think that approach would be preferable?

Reply:
Added new class SnowflakeLoggedFeatureNotSupportedException class

Reply:
Yes, looks better.